lib/dor/models/preservable.rb

```diff
@@ -11,11 +11,9 @@ module Preservable
 
     def build_provenanceMetadata_datastream(workflow_id, event_text)
       workflow_provenance = create_workflow_provenance(workflow_id, event_text)
-      dsname = 'provenanceMetadata'
-      ds = datastreams[dsname]
-      ds.label = 'Provenance Metadata' unless datastreams.keys.include?(dsname)
+      ds = datastreams['provenanceMetadata']
+      ds.label ||= 'Provenance Metadata'
       ds.ng_xml = workflow_provenance
-      ds.content = ds.ng_xml.to_s
```

Comment:
I've pushed up https://github.com/sul-dlss/dor-services/pull/258, which should deal with all of those.. Next week, it's probably worth doing some testing in argo and other places to see if we can find (and then properly fix..) any issues.